bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitBaseAccessoryHandler.java

```diff
@@ -0,0 +1,874 @@
+    private final Map<Long, Accessory> accessories = new ConcurrentHashMap<>();
+    private final HomekitKeyStore keyStore;
+
+    private boolean isConfigured = false;
+    private int connectionAttemptDelay = MIN_CONNECTION_ATTEMPT_DELAY_SECONDS;
+
+    private volatile @Nullable ScheduledFuture<?> connectionAttemptTask;
+    private volatile @Nullable CharacteristicReadWriteClient rwService;
+    private volatile @Nullable IpTransport ipTransport;
+    private volatile @Nullable ScheduledFuture<?> refreshTask;
+    private volatile @Nullable Future<?> manualRefreshTask;
+
+    private @NonNullByDefault({}) Long accessoryId;
+
+    protected static final Gson GSON = new Gson();
+
+    /**
+     * Maps of evented and polled Characteristics.
+     * The maps are keyed on the unique "aid,iid" combination to prevent duplicate entries.
+     */
+    protected static final String AID_IID_FORMAT = "%s,%s";
+    protected final Map<String, Characteristic> eventedCharacteristics = new ConcurrentHashMap<>();
+    protected final Map<String, Characteristic> polledCharacteristics = new ConcurrentHashMap<>();
+
+    protected final HomekitTypeProvider typeProvider;
+    protected final TranslationProvider i18nProvider;
+    protected final Bundle bundle;
+
+    protected boolean isBridgedAccessory = false;
+    protected final Throttler throttler = new Throttler();
+
+    /**
+     * A helper class that runs a {@link Callable} and enforces a minimum delay between calls.
+     * This is to avoid overwhelming accessories with too many requests in a short time.
+     */
+    private class Throttler {
+        private static final Duration MIN_INTERVAL = Duration.ofSeconds(2);
+        private @Nullable Instant notBeforeInstant = null;
+
+        /**
+         * Calls the given task. The method is synchronized to ensure that only one HTTP call is
+         * executed at a time. It calculates the required delay based on the last call time and
+         * sleeps if necessary. And it updates the notBeforeInstant after each call to enforce the
+         * delay. It initializes notBeforeInstant if required.
+         *
+         * @param task the task to be called
+         * @return the String result of the task
+         * @throws Exception the compiler us to handle any exception, but will actually be more specific
+         */
+        public synchronized String call(Callable<String> task) throws Exception {
+            try {
+                Instant next = notBeforeInstant;
+                if (next == null) {
+                    notBeforeInstant = next = Instant.now().plus(MIN_INTERVAL);
+                }
+                long delay = Duration.between(Instant.now(), next).toMillis();
+                if (delay > 0) {
+                    delay = Math.min(delay, MIN_INTERVAL.toMillis());
+                    logger.trace("{} throttling call for {} ms to respect minimum interval", thing.getUID(), delay);
+                    Thread.sleep(delay);
```

Comment:
Correct me if I'm wrong, but I think `Thread.sleep` supports `Duration`, so I believe it can be simplified to:
```suggestion
                Duration delay = Duration.between(Instant.now(), next);
                if (!delay.isNegative() && !delay.isZero()) {
                    Duration sleepDuration = delay.min(MIN_INTERVAL);
                    logger.trace("{} throttling call for {} to respect minimum interval", thing.getUID(), sleepDuration);
                    Thread.sleep(sleepDuration);
```